modules/pinger/src/main/java/org/hawkular/component/pinger/PingDestination.java

```diff
@@ -16,74 +16,165 @@
+
 /**
  * A destination for pinging.
  *
  * @author Heiko W. Rupp
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
  */
 public class PingDestination {
+    public static final String URL_TYPE = "URL";
+
+    public enum ResourceField {
+        url, method
+    };
+
+    public static boolean isUrl(Resource r) {
+        return URL_TYPE.equals(r.getType().getId());
+    }
+
+    public static PingDestination from(Resource r) {
+        Map<String, Object> props = r.getProperties();
+        return new PingDestination(r.getTenantId(), r.getEnvironmentId(), r.getId(),
+                (String) props.get(ResourceField.url.name()), (String) props.get(ResourceField.method.name()));
+    }
 
     /** The default method {@value} */
     public static final String DEFAULT_METHOD = "GET";
 
-    final String resourceId;
-    final String url;
-    final String method;
+    private final String tenantId;
+    private final String environmentId;
+    private final String resourceId;
+    private final String url;
+    private final String method;
 
     /**
      * Creates a new {@link PingDestination} using the default method {@value #DEFAULT_METHOD}.
      *
-     * @param resourceId the resourceId of this destination
-     * @param url the url to ping
+     * @param tenantId the owner of the present {@link PingDestination}
+     * @param environmentId the environment (test/live) the present {@link PingDestination} belongs to
+     * @param resourceId the resourceId of this destination as taken from Hawkular Inventory
+     * @param url the URL to ping
      */
-    public PingDestination(String resourceId, String url) {
-        this(resourceId, url, DEFAULT_METHOD);
+    public PingDestination(String tenantId, String environmentId, String resourceId, String url) {
+        this(tenantId, environmentId, resourceId, url, DEFAULT_METHOD);
     }
 
     /**
      * Creates a new {@link PingDestination}
      *
-     * @param url the url to ping
+     * @param tenantId the owner of the present {@link PingDestination}
+     * @param environmentId the environment (test/live) the present {@link PingDestination} belongs to
+     * @param resourceId the resourceId of this destination as taken from Hawkular Inventory
+     * @param url the URL to ping
      * @param method the HTTP method to use in the ping request or null to use the default method
-     *               {@value #DEFAULT_METHOD}
+     *        {@value #DEFAULT_METHOD}
      */
-    public PingDestination(String resourceId, String url, String method) {
+    public PingDestination(String tenantId, String environmentId, String resourceId, String url, String method) {
+        this.tenantId = tenantId;
+        this.environmentId = environmentId;
         this.resourceId = resourceId;
         this.url = url;
         this.method = method == null ? DEFAULT_METHOD : method;
     }
 
+    /** Compares only {@link #resourceId} for performance reasons.
```

Comment:
This is technically incorrect, because resourceId is guaranteed to be unique only within its environment. I.e. in current situation, two tenants could create a resource with the same id (not sure if that's possible) and they would be claimed equal.